cli/lib/commands/init.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+module.exports = create;
+
+const readline = require('readline');
+const {
+  createPackageJson,
+  createToml,
+  fileExists,
+  isValidSemver,
+  writeFile
+} = require('../utils');
+
+const REJECTION_MSGS = {
+  REQUIRED: 'This is required',
+  validateName: "Sorry, that's not a valid name",
+  validateVersion: "Sorry, that's not a valid semver version"
+};
+
+const VALID_YES_NO = {
+  y: 1,
+  yes: 1,
+  n: 0,
+  no: 0
+};
+
+const QUESTIONS = {
+  TOML_EXISTS: {
+    MESSAGE: 'Package.toml already exists. Continue (Y/N)?: ',
+    REQUIRED: true
+  },
+  NAME: {
+    MESSAGE: 'Name: ',
+    REQUIRED: true
+  },
+  VERSION: {
+    MESSAGE: 'Version: ',
+    REQUIRED: true
+  },
+  ENTRY: {
+    MESSAGE: 'Entry: ',
+    REQUIRED: false
+  },
+  TYPE: {
+    MESSAGE: 'Type :',
+    REQUIRED: false
+  },
+  LICENSE: {
+    MESSAGE: 'License: ',
+    REQUIRED: false
+  },
+  DESCRIPTION: {
+    MESSAGE: 'Description: ',
+    REQUIRED: false
+  },
+  HOMEPAGE: {
+    MESSAGE: 'Homepage: ',
+    REQUIRED: false
+  },
+  AUTHOR: {
+    MESSAGE: 'Author: ',
+    REQUIRED: false
+  },
+  REPOSITORY: {
+    MESSAGE: 'Repository: ',
+    REQUIRED: false
+  }
+};
+
+/**
+ * Returns string for the path to write Package.toml
+ */
+function tomlLocation() {
+  return `${process.cwd()}/Package.toml`;
+}
+
+/**
+ * Returns boolean based upon whether a user's answer is in the VALID_YES_NO hash
+ */
+function validateYesNo(ans) {
+  if (!ans) {
+    return false;
+  }
+
+  return VALID_YES_NO[ans] !== undefined;
+}
+
+/**
+ * Validates the name provided by the user
+ */
+function validateName(name) {
+  // TODO: Uese Regex for "name@registry-domain.dev/package_name"?
+  return name && name.length > 0;
+}
+
+/**
+ * Validates the version provided by the user
+ */
+function validateVersion(version) {
+  return isValidSemver(version);
+}
+
+/**
+ * Used to provide consistent casing for hash lookups
+ */
+function lowercase(str) {
+  return str.toLowerCase();
+}
+
+/**
+ * Minimal prompt for asking users for input
+ */
+function ask(question, rl) {
+  return new Promise((resolve, reject) => {
+    rl.question(question, answer => resolve(answer));
+  }).catch(e => {
+    console.error(e);
```

Comment:
Is this thing able to throw an error? What do you need to do for it to throw one?

Reply:
There lies an edge case somewhere where this promise might get rejected. If an error does happen, a user should be able to open an issue with the log rather than just saying, "it doesn't work"

Reply:
Right. But the error will be thrown with a message and a stack anyway. I think you don't need this catch here.